src/library/scala/Enumeration.scala

```diff
@@ -188,7 +188,13 @@ abstract class Enumeration (initial: Int) extends Serializable {
   protected final def Value(i: Int, name: String): Value = new Val(i, name)
 
   private def populateNameMap(): Unit = {
-    val fields: Array[JField] = getClass.getDeclaredFields
+    @tailrec def getFields(clazz: Class[_], acc: Array[JField]): Array[JField] = {
+      if (clazz.getSuperclass == null)
+        acc ++ clazz.getDeclaredFields
+      else
+        getFields(clazz.getSuperclass, acc ++ clazz.getDeclaredFields)
+    }
+    val fields = getFields(getClass, Array.empty)
```

Comment:
It doesn't matter because Enumeration doesn't matter, but starting with an empty array is suboptimal. I see now `++` means allocate and copy. Usually there's a default size and then resize to actual when done. But here 99.99% is just one go. I'd suggest either `null` or `getFields(getClass, getclass.getDeclaredFields)`.